src/pub.rs

```diff
@@ -158,4 +161,38 @@ impl Socket for PubSocket {
         util::peer_connected(raw_socket, self.backend.clone()).await;
         Ok(())
     }
+
+    fn binds(&self) -> &[Endpoint] {
+        &self.binds
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::util::tests::{
+        test_bind_to_any_port_helper, test_bind_to_unspecified_interface_helper,
```

Comment:
`test_bind_to_any_port_helper` and the interface helper are only used by test cases, but they will be used by many different socket types. Let me know if this approach of putting them in the util::tests module is idiomatic.